collect_app/src/main/java/org/odk/collect/android/activities/InstanceUploaderList.java

```diff
@@ -132,55 +141,74 @@ public void denied() {
         });
     }
 
-    private void init() {
-        instancesDao = new InstancesDao();
-        uploadButton = findViewById(R.id.upload_button);
-        uploadButton.setOnClickListener(new View.OnClickListener() {
-
-            @Override
-            public void onClick(View v) {
-
-                String transport = (String) GeneralSharedPreferences.getInstance().get(KEY_SUBMISSION_TRANSPORT_TYPE);
-                if (!transport.equalsIgnoreCase(getString(R.string.transport_type_value_sms))) {
-
-                    ConnectivityManager connectivityManager = (ConnectivityManager) getSystemService(
-                            Context.CONNECTIVITY_SERVICE);
-                    NetworkInfo ni = connectivityManager.getActiveNetworkInfo();
+    /**
+     * Determines how an upload should be handled by checking the transport being used.
+     * If the transport is set to SMS or Internet then either is used respectively else it's
+     * "Both" so the button IDs drive the decision.
+     *
+     * @param button that triggers an upload
+     */
+    @OnClick({R.id.upload_button, R.id.sms_upload_button})
+    public void onUploadButtonsClicked(Button button) {
+        String transport = (String) GeneralSharedPreferences.getInstance().get(KEY_SUBMISSION_TRANSPORT_TYPE);
+        if (transport.equalsIgnoreCase(getString(R.string.transport_type_value_internet)) || button.getId() == R.id.upload_button) {
```

Comment:
I think there are some inconsistencies between `equals` and `equalsIgnoreCase` in different parts of the code. Maybe it's worth introducing a `isInternetTransportSelected`, `isSMSTransportSelected`, etc methods to make sure usage is consistent and reduce the cognitive load of some of these lines.

Reply:
Didn't you want to use the enum class you implemented here also? I think you forgot about it.

Reply:
Oh yes, thanks.